src/chainstate/stacks/index/bits.rs

```diff
@@ -310,12 +318,21 @@ pub fn read_nodetype<F: Read + Seek>(
 ///   node hash      id  ptrs & ptr data      path
 ///
 /// X is fixed and determined by the TrieNodeType variant.
-/// Y is variable, but no more than TriePath::len()
-pub fn read_nodetype_at_head<F: Read>(
+/// Y is variable, but no more than TriePath::len().
+///
+/// If `read_hash` is false, then the contents of the node hash are undefined.
+#[inline]
+pub fn read_nodetype_at_head<F: Read + Seek>(
```

Comment:
This interface seems more dangerous than necessary.

Could we have two method variants instead:

```
fn read_nodetype_at_head<F: Read + Seek>(f: &mut F, ptr_id: u8) -> Result<(TrieNodeType, TrieHash), Error>;
fn read_nodetype_at_head_no_hash<F: Read + Seek>(f: &mut F, ptr_id: u8) -> Result<TrieNodeType, Error>;
```

And maybe these two methods are just wrappers around a (non-public) inner implementation?

That way, there's never a method returning data that shouldn't be read.

Reply:
Sure, can do.